kayak_core/src/widget_manager.rs

```diff
@@ -248,6 +261,58 @@ impl WidgetManager {
         morphorm::layout(&mut self.layout_cache, &self.node_tree, &self.nodes);
     }
 
+    fn create_primitive(
+        &mut self,
+        id: Index,
+        styles: &mut Style,
+        assets: &mut Assets,
+    ) -> RenderPrimitive {
+        let mut render_primitive = RenderPrimitive::from(&styles.clone());
+
+        match &mut render_primitive {
+            RenderPrimitive::Text {
+                parent_size,
+                content,
+                font,
+                size,
+                line_height,
+                ..
+            } => {
+                // --- Bind to Font Asset --- //
+                let asset = assets.get_asset::<KayakFont, _>(font.clone());
+                self.bind(id, &asset);
+
+                if let Some(font) = asset.get() {
+                    if let Some(parent_id) = self.get_valid_parent(id) {
+                        if let Some(parent_layout) = self.get_layout(&parent_id) {
```

Comment:
Resolved by making the `render` method recursive (up to a max of 2). While rendering, if we add to the `dirty_render_nodes`, we can force the render method to be recursively invoked. This allows all widgets that rely on some layout to be rendered properly (prevents flickering for widgets that can be properly rendered within that recursion limit— which is most widgets).

If a widget can't be rendered properly within the recursion limit, it will be handled the next frame.